expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,837 @@
+                    reversals=list(self._reversals),
+                    x=list(self._x))))
+
+    def _stop_here(self):
+        if self._stop_rule.lower() == 'reversals':
+            self._n_stop = self._n_reversals
+        elif self._stop_rule.lower() == 'trials':
+            self._n_stop = self._n_trials
+        return self._n_stop >= self._stop_criterion
+
+    def _step_index(self):
+        if self._change_rule.lower() == 'reversals':
+            self._n_change = self._n_reversals
+        elif self._stop_rule.lower() == 'trials':
+            self._n_change = self._n_trials
+        return np.where(self._n_change >= self._change_criteria)[0][-1]
+
+    @property
+    def _current_step_size_up(self):
+        return self._step_size_up[self._step_index()]
+
+    @property
+    def _current_step_size_down(self):
+        return self._step_size_down[self._step_index()]
+
+    # =========================================================================
+    # Define all the public properties
+    # =========================================================================
+    @property
+    def up(self):
+        return self._up
+
+    @property
+    def down(self):
+        return self._down
+
+    @property
+    def step_size_up(self):
+        return self._step_size_up
+
+    @property
+    def step_size_down(self):
+        return self._step_size_down
+
+    @property
+    def stop_criterion(self):
+        return self._stop_criterion
+
+    @property
+    def stop_rule(self):
+        return self._stop_rule
+
+    @property
+    def start_value(self):
+        return self._start_value
+
+    @property
+    def x_min(self):
+        return self._x_min
+
+    @property
+    def x_max(self):
+        return self._x_max
+
+    @property
+    def stopped(self):
+        """Has the tracker stopped
+        """
+        return self._stopped
+
+    @property
+    def x(self):
+        """The staircase
+        """
+        return self._x
+
+    @property
+    def x_current(self):
+        """The current level
+        """
+        return self._x_current
+
+    @property
+    def responses(self):
+        """The response history
+        """
+        return self._responses
+
+    @property
+    def n_trials(self):
+        """The number of trials so far
+        """
+        return self._n_trials
+
+    @property
+    def n_reversals(self):
+        """The number of reversals so far
+        """
+        return self._n_reversals
+
+    @property
+    def reversals(self):
+        """The reversal history (0 where there was no reversal)
+        """
+        return self._reversals
+
+    @property
+    def reversal_inds(self):
+        """The trial indices which had reversals"""
+        return np.where(self._reversals)[0]
+
+    # =========================================================================
+    # Display functions
+    # =========================================================================
+    def plot(self):
```

Comment:
ideally would like this to have an argument `ax=None` so it could plot on an existing axes object if provided.  Could be a later PR if you're sick of this one though.  :sweat_smile: